connector/datadogconnector/connector.go

```diff
@@ -181,14 +186,16 @@ func (c *traceToMetricConnector) enrichStatsPayload(stats *pb.StatsPayload) {
 	for _, stat := range stats.Stats {
 		if stat.ContainerID != "" {
 			if tags, ok := c.containerTagCache.Get(stat.ContainerID); ok {
-				tagList := tags.(map[string]struct{})
+
+				tagList := tags.(*sync.Map)
 				for _, tag := range stat.Tags {
-					tagList[tag] = struct{}{}
-				}
-				stat.Tags = make([]string, 0, len(tagList))
-				for tag := range tagList {
-					stat.Tags = append(stat.Tags, tag)
+					tagList.Store(tag, struct{}{})
 				}
+				stat.Tags = make([]string, 0)
```

Comment:
Why remove the preallocation?

Reply:
`Sync.Map` doesn't provide a size method; we must iterate to get the size.

Reply:
Oki, makes sense